community/io/src/main/java/org/neo4j/io/mem/GrabAllocator.java

```diff
@@ -162,8 +166,13 @@ private void initCause( NativeMemoryAllocationRefusedError error, OutOfMemoryErr
     protected synchronized void finalize() throws Throwable
     {
         super.finalize();
-        Grab current = grabs;
+        close();
+    }
 
+    @Override
+    public void close()
+    {
+        Grab current = grabs;
```

Comment:
The tests that need this could also call `finalize` directly. The reason we free in `finalize` is because we rely on the GC to tell us when there are no more page cursors that can access any native memory. Having a `close` method makes it look like a mistake that we are not calling it anywhere in the product code (except for this finalizer). If you insist on having it, then we should strengthen the documentation around the particular way that we are keeping track of the life cycle of the native memory in the page cache, and also the method should be `synchronized` to match its siblings in this class.

Reply:
Definitely add a comment for future victims to digest :)